parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -262,12 +313,17 @@ public void writeTo(OutputStream out) throws IOException {
     }
 
     @Override
+    public byte[] getBytesUnsafe() {
+      return getBytes();
```

Comment:
Maybe. There are quite a few methods that call this internally, like `hashCode` and `equals`. If we are hiding the fact that a copy is made internally, then the caller doesn't know to cache anything.

But, the solution is probably to improve the internal implementations. For example, `hashCode` might be improved to avoid the need for a copy.